### ISSUE TITLE
Error in pairing while running hicup_mapper

### ISSUE THREAD

**Issue by Giuseppe1995** [StevenWingett/HiCUP#66]
Hi,
I am trying to run the hicup pipeline on some [test data](http://www.bioinformatics.babraham.ac.uk/projects/hicup/test_dataset.tar.gz). I correctly truncated reads via ´hicup_truncater´, generating the files test_dataset1.trunc.fastq and test_dataset2.trunc.fastq (I am assuming that they refer to the forward and reverse readset, respectively). However, I'm getting an error when I run `hicup_mapper`

` hicup_mapper --bowtie2 $BOWTIE2 --index $BWT_IDX --format Illumina_1.5 --threads $NT test_dataset1.trunc.fastq test_dataset2.trunc.fastq`

> Detecting R automatically
Found R at '/srv/ngsdata/dalteriog/Tools/miniconda3/bin/R'
Mapping with HiCUP Mapper v0.8.3
Using aligner 'bowtie2'
Mapping test_dataset1.trunc.fastq
Mapping test_dataset2.trunc.fastq
Mapping complete
Pairing files with HiCUP Mapper v0.8.3
Pairing test_dataset1.map.sam and test_dataset2.map.sam
Can't read './test_dataset1.map.sam' : No such file or directory at /srv/ngsdata/dalteriog/Tools/miniconda3/bin/hicup_mapper line 552.
Pairing complete
Could not delete './.test_dataset1.map.sam'
Could not delete './.test_dataset2.map.sam'
During startup - Warning message:
Setting LC_CTYPE failed, using "C" 
Error in read.table(file = file, header = header, sep = sep, quote = quote,  : 
  no lines available in input
Calls: read.delim -> read.table
Execution halted
Could not produce hicup_mapper summary bar chart: /srv/ngsdata/dalteriog/Tools/miniconda3/bin/Rscript /srv/ngs/analysis/dalteriog/Tools/miniconda3/share/hicup-0.8.3-0/r_scripts/hicup_mapper_summary.r ./ ./hicup_mapper_summary_uEOmJNsSPk_15-55-06_29-03-2022.txt: No such file or directory at /srv/ngsdata/dalteriog/Tools/miniconda3/bin/hicup_mapper line 240.

Here follows my variables explanation: 
$BOWTIE2 --> path to bowtie2 executable
$BWT_IDX --> path to bowtie2 index (without trailing .X.bt2)
$NT --> 8

I cannot develop in perl, but I tried to take a look at the script in the line   552:

`cat /srv/ngsdata/dalteriog/Tools/miniconda3/bin/hicup_mapper | awk 'NR==552{print}'`
        open( FORWARD, $fileForward ) or die "Can't read \'$fileForward\' : $!";

I am assuming that the problem is in the pairing of the produces mapping files, probably because they are not produced at all.
Is there something I am forgetting?

**Comment by StevenWingett** [comment]
Hi,

Sorry to learn you are experiencing difficulties with HiCUP.  Have any BAM/SAM files been generated?  Do they contain any alignments?

I have a few suggestions to try to solve the problem.

1). Try allocating more RAM to the job for it to run.  (This is excessive, but try 20 Gb RAM - does that work?)

2) If not, are you able to map the FASTQ files to the reference genome using Bowtie2 only?  This tells you whether Bowtie2 is working on your system.

3) Could you try installing the software without conda.  I do not maintain the conda install myself and cannot debug HiCUP installed using conda.  For more details on installation, please go to: https://www.bioinformatics.babraham.ac.uk/projects/hicup/read_the_docs/html/index.html#hicup-quick-start-guide

HiCUP is very easy to install.  Basically you just need to download the Perl scripts and run them.  

What system are you running this on?

Thanks.

**Comment by Giuseppe1995** [comment]
Oh, the problem stood with bowtie2! I run `bowtie2 --help` and I got this error

> /srv/ngsdata/dalteriog/Tools/miniconda3/bin/bowtie2-align-s: error while loading shared libraries: libtbb.so.2: cannot open shared object file: No such file or directory
(ERR): Description of arguments failed!
Exiting now ...

which I resolved thanks to [this issue](https://github.com/TaliaferroLab/LABRAT/issues/3). This was quite odd for me because I already run the whole hicup pipeline, but I was on another machine, on the same server.
Thank you so much!
